### PR TITLE
add checks for formatting and validating terraform

### LINKED ISSUE #1807: Add CI check to format and validate terraform files

### PR DESCRIPTION
**Pull request by shubha-rajan**
Fixes #1807

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -97,6 +97,21 @@ jobs:
...
+    steps:
+      - name: check out pr branch
+        uses: actions/checkout@v3
+      - name: generate bastion key
+        run:  ssh-keygen -q -t rsa -b 4096 -N '' -f "${HOME}/.ssh/bastion"
+      - name: setup terraform
+        uses: hashicorp/setup-terraform@v1
+      # as we add more templates, we will add checks to validate them. for now, we only have one
+      - name: terraform init azure_saml_ses 
+        run: terraform -chdir=cloud/azure/templates/azure_saml_ses init -backend=false
```
Comment (goldblatt) [inline .github/workflows/pr.yaml:111]: ooo does this work like this with backend=false

Comment (goldblatt) [reply .github/workflows/pr.yaml:111]: lol i should do this instead of the overrides

**Comment by shubha-rajan** [comment]
> I think ideally we'd have something that confirms like we generate the setup.auto.tfvars correctly as well but i this is dope!

Yeah I considered adding a plan step there too, but then we have to deal with credentials and authentication for azure and I didn't want to do that. 

you can see that the validation failed before I added the bastion key, and the formatting failed before I formatted the GCP files. So what's already in the PR is working.

File: .github/workflows/pr.yaml (hunk 1 above)
Comment (shubha-rajan) [reply .github/workflows/pr.yaml:111]: https://www.terraform.io/cli/commands/validate
"Validation requires an initialized working directory with any referenced plugins and modules installed. To initialize a working directory for validation without accessing any configured remote backend, use:
```
terraform init -backend=false
```
To verify configuration in the context of a particular run (a particular target workspace, input variable values, etc), use the terraform plan command instead, which includes an implied validation check."

**Pull request merged**